### PR TITLE
Findbugs fixes

### PR DESCRIPTION
**Pull request by rozza**
[Review on Reviewable](https://reviewable.io/reviews/mongodb/mongo-java-driver/313)

### CODE REVIEW

File: config/findbugs-exclude.xml [hunk 1]
```diff
@@ -8,16 +8,14 @@
     </Match>
 
     <Match>
-        <Package name="com.mongodb">
-            <Bug pattern="EQ_DOESNT_OVERRIDE_EQUALS"/>  <!-- Deliberately ignoring this, as many BSONObject subclasses don't do it -->
```
Comment (evanchooly) [inline config/findbugs-exclude.xml]: I guess findbugs doesn't have an xsd to validate against?

File: config/findbugs-exclude.xml [hunk 2]
```diff
@@ -26,9 +24,8 @@
         <Bug pattern="OBL_UNSATISFIED_OBLIGATION"/>
     </Match>
     <Match>
-        <Class name="com.mongodb.DocumentCodec">
+        <Class name="com.mongodb.DocumentCodec" />
             <Bug pattern="NM_SAME_SIMPLE_NAME_AS_SUPERCLASS"/>
```
Comment (evanchooly) [inline config/findbugs-exclude.xml]: indentation is off now, fwiw.

**Comment by evanchooly** [comment]
:+1:

File: config/findbugs-exclude.xml (hunk 2 above)
Comment (jyemin) [inline config/findbugs-exclude.xml]: It's worth something.  Should be fixed. ;)

**Comment by rozza** [comment]
Updated to fix findbugsTest issues

**Comment by rozza** [comment]
Done

File: driver/src/main/org/bson/LazyBSONObject.java [hunk 3]
```diff
@@ -170,8 +170,7 @@ Object readValue(final BsonBinaryReader reader) {
             case BINARY:
                 BsonBinary binary = reader.readBinaryData();
                 byte binaryType = binary.getType();
-                if (binaryType == BsonBinarySubType.BINARY.getValue()
-                    || binaryType == BsonBinarySubType.BINARY.getValue()) {
+                if (binaryType == BsonBinarySubType.BINARY.getValue()) {
```
Comment (rozza) [inline driver/src/main/org/bson/LazyBSONObject.java]: Could this have supposed to have been:

```
binaryType == BsonBinarySubType.BINARY.getValue()
-                    || binaryType == BsonBinarySubType.OLD_BINARY.getValue()
```

**Comment by rozza** [comment]
More updates @jyemin @evanchooly

**Comment by jyemin** [comment]
LGTM after the obvious squashing is done.

**Comment by rozza** [comment]
Squashed, rebased and merged

**Pull request closed without merging**